src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -364,9 +372,18 @@ public static IntPtr AllocateTypeAssociatedMemory(Type type, int size)
     }
     // Helper class to assist with unsafe pinning of arbitrary objects.
     // It's used by VM code.
+    [StructLayout(LayoutKind.Explicit)]
     internal sealed class RawData
     {
+        [FieldOffset(0)]
```

Comment:
I do not see the improvement here. Is this just a micro-optimization to componsate for missing JIT optimization? I would much preffer to fix the JIT instead of this.